src/language/markdown-source-code.js

```diff
@@ -23,6 +30,104 @@ import { VisitNodeStep, TextSourceCodeBase } from "@eslint/plugin-kit";
 /** @typedef {import("@eslint/core").ParseResult<RootNode>} ParseResult */
 /** @typedef {import("@eslint/core").SourceLocation} SourceLocation */
 /** @typedef {import("@eslint/core").SourceRange} SourceRange */
+/** @typedef {import("@eslint/core").FileProblem} FileProblem */
+/** @typedef {import("@eslint/core").DirectiveType} DirectiveType */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const commentParser = new ConfigCommentParser();
+const configCommentStart =
+	/<!--\s*(eslint(?:-enable|-disable(?:(?:-next)?-line)?)?)/u;
+const htmlComment = /<!--(.*?)-->/gsu;
+
+/**
+ * Represents an inline config comment in the source code.
+ */
+class InlineConfigComment {
+	/**
+	 * The comment text.
+	 * @type {string}
+	 */
+	value;
+
+	/**
+	 * The position of the comment in the source code.
+	 * @type {SourceLocation}
+	 */
+	position;
+
+	/**
+	 * Creates a new instance.
+	 * @param {Object} options The options for the instance.
+	 * @param {string} options.value The comment text.
+	 * @param {SourceLocation} options.position The position of the comment in the source code.
+	 */
+	constructor({ value, position }) {
+		this.value = value.trim();
+		this.position = position;
+	}
+}
+
+/**
+ * Extracts inline configuration comments from an HTML node.
+ * @param {HTMLNode} node The HTML node to extract comments from.
+ * @returns {Array<InlineConfigComment>} The inline configuration comments found in the node.
+ */
+function extractInlineConfigCommentsFromHTML(node) {
+	if (!configCommentStart.test(node.value)) {
+		return [];
+	}
+	const comments = [];
+
+	let match;
+
+	while ((match = htmlComment.exec(node.value))) {
+		if (configCommentStart.test(match[0])) {
+			const comment = match[0];
+
+			// calculate location of the comment inside the node
+			const start = {
+				...node.position.start,
+			};
+
+			const end = {
+				...node.position.start,
+			};
+
+			const {
+				lineOffset: startLineOffset,
+				columnOffset: startColumnOffset,
+			} = findOffsets(node.value, match.index);
+
+			start.line += startLineOffset;
+			start.column += startColumnOffset;
+			start.offset += match.index;
+
+			const commentLineCount = comment.split("\n").length - 1;
+
+			end.line = start.line + commentLineCount;
+			end.column =
+				commentLineCount === 0
+					? start.column + comment.length
+					: comment.length - comment.lastIndexOf("\n") - 1;
```

Comment:
```suggestion
					: comment.length - comment.lastIndexOf("\n");
```

As the columns are 1-based in this language.